rwengine/src/script/ScriptFunctions.hpp

```diff
@@ -43,13 +71,59 @@ inline bool objectInBounds(GameObject* object,
 	        p.y <= max.y && p.z <= max.z);
 }
 template <class Tobj, class Tvec>
-bool objectInArea(const ScriptArguments& args, Tobj* object, const Tvec& min, const Tvec& max, bool marker)
+bool objectInArea(const ScriptArguments& args, const Tobj& object, const Tvec& min, const Tvec& max, bool marker, bool preconditions = true)
+{
+	if (marker) {
+		auto center = (min+max)/2.f;
+		auto radius = glm::abs(max-min);
+		drawAreaIndicator(args, center, radius);
+	}
+	return preconditions && objectInBounds(object, glm::min(min, max), glm::max(min, max));
+}
+
+inline bool objectInSphere(GameObject* object,
+                           const ScriptVec2& center,
+                           const ScriptVec2& radius)
+{
+	auto p = glm::vec2(object->getPosition());
+	p = glm::abs(p - center);
+	return p.x <= radius.x && p.y <= radius.y;
+}
+inline bool objectInSphere(GameObject* object,
+                           const ScriptVec3& center,
+                           const ScriptVec3& radius)
+{
+	auto p = object->getPosition();
+	p = glm::abs(p - center);
+	return p.x <= radius.x && p.y <= radius.y && p.z <= radius.z;
+}
+template <class Tobj, class Tvec>
+inline bool objectInRadius(const ScriptArguments& args, const Tobj& object, const Tvec& center, const Tvec& radius, bool marker, bool preconditions = true)
+{
+	if (marker) {
+		drawAreaIndicator(args, center, radius);
+	}
+	return preconditions && objectInSphere(object, center, radius);
+}
+template <class Tvec>
+inline bool objectInRadiusNear(const ScriptArguments& args, GameObject* object, GameObject* near, const Tvec& radius, bool marker, bool preconditions = true)
 {
+	Tvec center (near->getPosition());
 	if (marker) {
-		RW_UNUSED(args);
-		RW_UNIMPLEMENTED("Area check marker");
+		drawAreaIndicator(args, center, radius);
 	}
-	return objectInBounds(object, min, max);
+	return preconditions && objectInSphere(object, center, radius);
+}
+
+template<class Tobj>
+inline bool objectInZone(const ScriptArguments& args, Tobj object, const ScriptString zone) {
+	auto zfind = args.getWorld()->data->zones.find(zone);
+	if( zfind != args.getWorld()->data->zones.end() ) {
+		auto& min = zfind->second.min;
+		auto& max = zfind->second.max;
+		return objectInBounds(object, min, max);
+	}
+	return false;
```

Comment:
Beyond the scope: This should probably be part of the object class in the future?